### PR TITLE
Batch sample project rename and other doc fixes

### PR DESCRIPTION
**Pull request by abhijitpai**
## Description
Rename Batch in Samples to TransactionalBatch.

General doc fixes across SDK to fix non-displayed / stray links in generated documentation:
> see and seealso with href need text for display. 
> seealso should be top level (not within remarks etc.) and isn't inlined into the documentation but shows up as a separate section. 
> exception needs cref. 
> list needs a type attribute.

## Type of change

Sample and documentation update.

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/CosmosClient.cs [hunk 1]
```diff
@@ -87,15 +85,13 @@ namespace Microsoft.Azure.Cosmos
...
-    /// <remarks>
     /// <seealso cref="CosmosClientOptions"/>
     /// <seealso cref="Fluent.CosmosClientBuilder"/>
-    /// <seealso href="https://docs.microsoft.com/azure/cosmos-db/performance-tips"/>
-    /// <seealso href="https://docs.microsoft.com/azure/cosmos-db/troubleshoot-dot-net-sdk"/>
-    /// <seealso href="https://docs.microsoft.com/azure/cosmos-db/distribute-data-globally" />
-    /// <seealso href="https://docs.microsoft.com/azure/cosmos-db/partitioning-overview" />
-    /// <seealso href="https://docs.microsoft.com/azure/cosmos-db/request-units" />
-    /// </remarks>
+    /// <seealso href="https://docs.microsoft.com/azure/cosmos-db/performance-tips">Performance Tips</seealso>
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/CosmosClient.cs]: These should be hotlinks so that we can change the URLs in the future if the documents get moved.

Comment (abhijitpai) [reply Microsoft.Azure.Cosmos/src/CosmosClient.cs]: Is there a standard protocol to get these created? Also, we can change the URLs here if they are no longer relevant, correct?

Comment (ealsur) [reply Microsoft.Azure.Cosmos/src/CosmosClient.cs]: We can use aka.ms links (they can be created on demand) if needed. But the issue is the same, if the URL changes in the future, someone needs to remember to update the short link, and here is more evident and anyone can do it.

File: Microsoft.Azure.Cosmos/src/Serializer/CosmosSerializer.cs [hunk 2]
```diff
@@ -16,18 +16,18 @@ public abstract class CosmosSerializer
...
+        /// <typeparam name="T">Any type passed to <see cref="Container"/>.</typeparam>
+        /// <param name="stream">The Stream response containing JSON from Cosmos DB.</param>
         /// <returns>The object deserialized from the stream.</returns>
         public abstract T FromStream<T>(Stream stream);
 
         /// <summary>
         /// Convert the object to a Stream. 
         /// The caller will take ownership of the stream and ensure it is correctly disposed of.
-        /// Stream.CanRead must be true https://docs.microsoft.com/dotnet/api/system.io.stream.canread?view=netcore-2.0
+        /// <see href="https://docs.microsoft.com/dotnet/api/system.io.stream.canread?view=netcore-2.0">Stream.CanRead</see> must be true.
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/Serializer/CosmosSerializer.cs]: I would remove the `view` as it is pointing to NET Core 2 only.
```suggestion
        /// <see href="https://docs.microsoft.com/dotnet/api/system.io.stream.canread">Stream.CanRead</see> must be true.
```

File: Microsoft.Azure.Cosmos/src/CosmosClient.cs (hunk 1 above)
Comment (abhijitpai) [reply Microsoft.Azure.Cosmos/src/CosmosClient.cs]: Thanks - will keep as is per above.

File: Microsoft.Azure.Cosmos/src/Batch/TransactionalBatch.cs [hunk 3]
```diff
@@ -5,20 +5,105 @@
...
+    ///         using (StreamReader streamReader = new StreamReader(operationResult.ResourceStream))
+    ///         {
+    ///             resultItems.Add(await streamReader.ReadToEndAsync());
+    ///         }
+    ///     }
+    /// }
+    /// ]]>
+    /// </code>
+    /// </example>
+    /// <seealso href="https://docs.microsoft.com/en-us/azure/cosmos-db/concepts-limits">Limits on TransactionalBatch requests</seealso>
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/Batch/TransactionalBatch.cs]: Remove "/en-us"

**Pull request merged**